doc/develop/tools/clion.rst

```diff
@@ -0,0 +1,270 @@
+profile for the project.
+
+#. Go to **Settings | Build, Execution, Deployment | CMake**.
+
+#. Select the **Zephyr-Debug preset** profile and set the **Enable profile**
+   checkbox.
+
+#. Delete the automatically created **Debug** profile. Alternatively,
+    move the **Zephyr-Debug preset** profile to the top of the list
+    to make it default for the project.
+
+#. Click **Apply** and close the dialog.
+
+   CMake project will be reloaded.
+
+Build the project
+*****************
+
+#. Check the top toolbar: the **Zephyr-Debug** profile and the
+   **zephyr_final** configuration should be selected.
+
+#. Click the hammer icon next to the configuration switcher.
+
+   When the build is finished, you can find build artifacts in
+   the ``hello_world\build`` folder.
+
+Note that other CMake targets like ``puncover`` or ``hardenconfig``
+can also be called at this point.
+
+Configure Zephyr parameters for debug
+*************************************
+
+#. In the configuration switcher, select **guiconfig** and click the hammer
+   icon.
+
+#. Use the GUI application to set the following flags:
+
+   .. code-block::
+
+      CONFIG_DEBUG_THREAD_INFO
+      CONFIG_THREAD_RUNTIME_STATS
+      CONFIG_DEBUG_OPTIMIZATIONS
+
+Enable RTOS integration
+***********************
+
+#. Go to **Settings | Build, Execution, Deployment | Embedded Development |
+   RTOS Integration**.
+
+#. Set the **Enable RTOS Integration** checkbox.
+
+   This option enables Zephyr tasks view during debugging. See
+   `Multi-threaded RTOS debug`_ in CLion web help for more information.
+
+   .. _Multi-threaded RTOS debug: https://www.jetbrains.com/help/clion/rtos-debug.html
+
+    You can leave the option set to **Auto**. CLion will detect Zephyr
+    automatically.
+
+Create an Embedded GDB Server configuration
+*******************************************
+
+In order to debug the Zephyr application, you need to create
+a run/debug configuration out of the Embedded GDB Server template.
+This configuration will call :ref:`west debugserver` under the hood
+and then use CLion's embedded GDB server to flash the target and
+start a debug session.
+
+#. In the configuration switcher, select **Edit Configurations**.
+
+#. Click "+" in the left-hand pane and select **Embedded GDB Server**.
+
+#. Configure the settings:
+
+    .. list-table::
+        :header-rows: 1
+
+        * - Option
+          - Value
+
+        * - **Name** (optional)
+          - Debug Zephyr
+
+        * - **Target**
+          - zephyr_final
+
+        * - **Executable binary**
+          - zephyr_final
+
+        * - **Debugger**
+          - Bundled GDB
+
+            *Note*: For non-ARM and non-x86 architectures, use a GDB executable
+            form Zephyr SDK. Make sure to pick a version with Python support
+            (for example, **riscv64-zephyr-elf-gdb-py**) and check that Python
+            is presented in the system ``PATH``.
```

Comment:
maybe use note admonition - it doesn't render 100% perfectly, but it might be worth using it nevertheless if you really want to make sure people don't miss out this piece of information.

```suggestion
            .. note:: For non-ARM and non-x86 architectures, use a GDB executable
               form Zephyr SDK. Make sure to pick a version with Python support
               (for example, **riscv64-zephyr-elf-gdb-py**) and check that Python
               is presented in the system ``PATH``.
```